app/javascript/src/components/availability/index.jsx

```diff
@@ -1,161 +1,198 @@
+        avail.tempID == event.tempID ? newEvent : avail
+      ));
+      updateAvailInfo(newAvails);
+    } else {
+      this.setState({
+        availData: event,
+        open: true,
+      });
     }
+
   };
 
-  renderKey = () => (
-    <div>
-      <Grid relaxed="very" padded>
-        <Grid.Row columns={3}>
-          <Grid.Column textAlign="center">
-            <h4> Unavailable </h4> <Icon color='red' name='cancel' size='large' />
-          </Grid.Column>
-          <Grid.Column textAlign="center">
-            <h4> Somewhat Available </h4> <Icon color='yellow' name='question circle' size='large' />
-          </Grid.Column>
-          <Grid.Column textAlign="center">
-            <h4> Available </h4> <Icon color='green' name='checkmark' size='large' />
-          </Grid.Column>
-        </Grid.Row>
-        <Grid.Row columns={1}>
-          <Grid.Column textAlign="center">
-            <i>Click icons below to select availability.</i>
-          </Grid.Column>
-        </Grid.Row>
-      </Grid>
-
-    </div>
-  );
+  eventPropGetter = ({ somewhat }) => ({
+    style: {
+      backgroundColor: somewhat ? '#CCCC00' : 'green',
+    },
+  });
+
+  // Modal functions
+  onOpen = () => this.setState({ open: true });
+  onClose = () => this.setState({ open: false });
 
   render() {
+    const {
+      availabilities,
+      fixed, signup,
+      deleteAvail, putAvail,
+    } = this.props;
+    const { availData, open, somewhat } = this.state;
+
+    // events for Calendar component
+    const events = availabilities.map(avail => ({
+      ...avail,
+      title: avail.somewhat ? 'Somewhat Available' : 'Available',
+      start: new Date(avail.start),
+      end: new Date(avail.end),
+    }));
+
+    const components = {
+      event: CustomEvent,
+    };
+
     return (
       <div>
-        {this.renderKey()}
-        <Grid columns='equal' celled >
-          <Grid columns={8} padded>
-          {days.map(day => (
-            <Grid.Column textAlign="center" key={day} color='green'>
-              <Grid.Row style={{ height: '10%' }}>{day}</Grid.Row>
-            </Grid.Column>
-          ))}
-          </Grid>
-          {times.map((time, row) => (
-            <Grid.Row key={time} style={{ height: '6.5%' }}>
-              <Grid.Column> {time} </Grid.Column>
-              {this.renderColumns(row)}
-            </Grid.Row>
-          ))}
-        </Grid>
+        {
+          !fixed &&
+          <SelectAvailType
+            value={somewhat}
+            handleChange={this.handleSomewhatChange}
+          />
+        }
+        {
+          !signup && !fixed &&
+          <ModalUpdate
+            open={open}
+            event={availData}
+            onOpen={this.onOpen}
+            onClose={this.onClose}
+            deleteAvail={deleteAvail}
+            putAvail={putAvail}
+          />
+        }
+        <DragDropCal
+          resizeable
+          popup
+          selectable={!fixed}
+          step={15}
```

Comment:
Love it!

Reply:
What exactly did you love here, just wondering? Also, I am noticing I need more comments. Lmk if you see anything that looks like it could use explanation.

Reply:
15 minute step